gbpservice/nfp/lifecycle_manager/modules/device_lifecycle_manager.py

```diff
@@ -0,0 +1,763 @@
+#from gbpservice.nfp.lifecycle_manager.compute.drivers import (
+#    nova_driver)
+'''from gbpservice.nfp.lifecycle_manager.drivers import (
+    haproxy_lifecycle_driver)
+from gbpservice.nfp.lifecycle_manager.drivers import (
+    vyos_lifecycle_driver)
+'''
+from neutron.common import rpc as n_rpc
+from neutron import context as n_context
+
+LOG = logging.getLogger(__name__)
+
+
+def rpc_init(controller, config):
+    rpcmgr = RpcHandler(config, controller)
+    agent = RpcAgent(
+        controller,
+        host=config.host,
+        topic=nsf_topics.NFP_CONFIGURATOR_DLCM_TOPIC,
+        manager=rpcmgr)
+    controller.register_rpc_agents([agent])
+
+
+def events_init(controller, config):
+    device_lifecycle_mgr_obj = DeviceLifeCycleManager(controller, config)
+    evs = [
+        Event(id='CREATE_NETWORK_FUNCTION_DEVICE',
+              handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_SPAWNING', handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_UP', handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_HEALTHY', handler=device_lifecycle_mgr_obj),
+        Event(id='CONFIGURE_DEVICE', handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_CONFIGURED', handler=device_lifecycle_mgr_obj),
+
+        Event(id='DELETE_NETWORK_FUNCTION_DEVICE',
+              handler=device_lifecycle_mgr_obj),
+        Event(id='DELETE_CONFIGURATION', handler=device_lifecycle_mgr_obj),
+        Event(id='DELETE_CONFIGURATION_COMPLETED',
+              handler=device_lifecycle_mgr_obj),
+        Event(id='DELETE_DEVICE', handler=device_lifecycle_mgr_obj),
+
+        Event(id='DEVICE_NOT_UP', handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_NOT_REACHABLE', handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_CONFIGURATION_FAILED',
+              handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_ERROR', handler=device_lifecycle_mgr_obj),
+        ]
+    controller.register_events(evs)
+
+
+def module_init(controller, config):
+    events_init(controller, config)
+    rpc_init(controller, config)
+    LOG.info(_("DLCM: module_init"))
+
+
+class RpcHandler(object):
+    RPC_API_VERSION = '1.0'
+
+    def __init__(self, conf, controller):
+        super(RpcHandler, self).__init__()
+        self.conf = conf
+        self._controller = controller
+        self.rpc_event_mapping = {'healthmonitor': ['DEVICE_HEALTHY',
+                                                   'DEVICE_NOT_REACHABLE'],
+                                  'interfaces':    ['DEVICE_CONFIGURED',
+                                                   'DEVICE_CONFIGURED_FAILED'],
+                                  'routes':        ['DEVICE_CONFIGURED',
+                                                   'DEVICE_CONFIGURED_FAILED'],
+                                  }
+
+    # RPC APIs status notification from Configurator
+    def get_network_function_config_info(self, context, notifications_data):
```

Comment:
Yes this is a response from the configurator.
Device lifecycle manager do/get RPC only from configurator, device lcm won't get RPCs from node_driver.